packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,543 @@
+
+#### 1. Composable API
+Composable API is a pattern where we have a set of components that can be composed together to create a table. This approach allows us to stitch together a table very similar to how you would do it with native HTML. However, it is also more verbose and requires more effort to create a table.
+
+```jsx
+// Mock data & functions
+
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const onSortChange = ({ headerKey, sortType }) => {
+  console.log({headerKey, sortType});
+};
+```
+
+
+```jsx
+// Composable Table
+
+<Table 
+    data={data} // This contains the actual data to be rendered in the table which would be retrieved from some API
+    selectionType='single' 
+    onSelectionChange={onSelectionChange} 
+    sortFunctions={sortFunctions} 
+    onSortChange={onSortChange}
+    isLoading={false}
+    cellDensity='normal'
+    showZebraStripes={true}
+    pagination={{
+      limit: 10,
+      offset: 0,
+    }}
+>
+{(tableData) => (
+  <>
+    <TableToolbar title="Users" itemsSelectedTitle={`${selectedUsersCount} Users selected`}>
+        <ToolbarActions>
+            <Button>Export</Button>
+        </ToolbarActions>
+    </TableToolbar>
+
+    <TableHeader isSticky={true}>
+      <TableHeaderRow>
+         <TableHeaderCell headerKey='firstName'>
+          First Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='lastName'>
+          Last Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='balance'>
+          Balance
+          <Tooltip content='Current bank balance'>
+            <InfoIcon size='medium' />
+          </Tooltip>
+        </TableHeaderCell>
+       </TableHeaderRow>
+     </TableHeader>
+
+    <TableBody>
+      {tableData.map((row) => (
+        <TableRow onClick={console.log} isDisabled={false}>
+          <TableCell>
+            {row.firstName}
+          </TableCell>
+          <TableCell>
+            {row.lastName}
+          </TableCell>
+          <TableCell>
+            {row.balance}
+            <Badge>Low</Badge>
+          </TableCell>
+        <TableRow>
+      )}
+    </TableBody>
+
+    <TableFooter>
+      <TableFooterRow>
+        <TableFooterCell columnStart='1' columnEnd='4'>
+          Total
+        </TableFooterCell>
+        <TableFooterCell>
+          {computedTotalBalance}
+        </TableFooterCell>
+      </TableFooterRow>
+    </TableFooter>
+
+    <TablePagination 
+      navigationType='compact' 
+      label='1-10 of 100'
+      labelPosition='left'
+      showRowCountPicker
+      rowCount='25'
+      onPaginationChange={console.log}
+      onRowCountPickerChange={console.log}
+      /> 
+   </>
+  )
+}
+</Table>
```

Comment:
Can you add an example of column grouping etc. which will cause verbosity issues ?

Reply:
Column grouping isn't what causes verbosity issues. Any table, even a simple one will be less verbose with compact API compared to composable API